Implement text class visualizer

MoodyMan04 opened the pull request.

MoodyMan04 committed:
text class created

created class for making text in rect
Default Font: Bauhaus93
Default Color: #daa520
Default Position: Vector(0, 0)

MoodyMan04 committed:
added getter and setter methods

added getter and setter methods to protect values of text

MoodyMan04 committed:
added comments

JeanVArdent reviewed (changes requested):
If you change values you have to recalc all of your values in the setter, also need to define colour in the constructor

MoodyMan04 committed:
Merge branch 'Visualizer' into implement-text-class-visualizer

MoodyMan04 committed:
Update text.py

- fixed constructor
- fixed type hinting
-  added reevaluating to created text when changing values in setters
- allow full support of possible Color parameters

JeanVArdent commented on Visualiser2/utils/text.py:
```diff
@@ -1,4 +1,114 @@
-    def __init__(self):
-        pass
\ No newline at end of file
+    def __init__(self, screen: pygame.Surface, text: str, font_size: int, font_name: str = 'bauhaus93', color: str | int | tuple(int, int, int, [int]) | list(int, int, int, [int]) | pygame.Color = pygame.Color('#daa520'), position: Vector = Vector(0, 0)):
```
Use optional

KingPhilip14 commented on Visualiser2/utils/text.py:
```diff
@@ -1,4 +1,114 @@
+
+    @font_size.setter
+    def font_size(self, font_size: int) -> None:
+        if font_size is None or not isinstance(font_size, str):
```
Checking for a string not an int

MoodyMan04 committed:
Update text.py

- fixed some formatting
- fixed wrong check: str -> int

JeanVArdent reviewed (approved):
lgtm

KingPhilip14 commented on Visualiser2/utils/text.py:
```diff
@@ -1,4 +1,114 @@
+    def __init__(self, screen: pygame.Surface, text: str, font_size: int, font_name: str = 'bauhaus93', color: str | int | tuple[int, int, int, Optional[int]] | list[int, int, int, Optional[int]] | pygame.Color = pygame.Color('#daa520'), position: Vector = Vector(0, 0)):
+        self.screen: pygame.Surface = screen  # assign screen used
+        self.text: str = text  # assign text used
+        self.font_size = font_size  # assign font size used
```
Type hinting?

MoodyMan04 committed:
font_size type hint

:eyes: ian...

KingPhilip14 reviewed (approved):
Looks good
